torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3705,17 +3763,43 @@ struct to_ir {
             range, sv->asValue(val_range, method), subscript_exprs));
       }
     } else {
-      // Desugars gather syntactic sugar foo[i]
-      Value* idx = emitExpr(subscript_exprs[0]);
-      Value* val = sv->asValue(val_range, method);
       AT_ASSERT(subscript_exprs.size() == 1);
+      Value* sliceable = sv->asValue(val_range, method);
+
+      auto subscript_sv = emitSugaredExpr(subscript_exprs[0], 1);
+      if (const auto psv =
+              dynamic_cast<PythonSliceValue*>(subscript_sv.get())) {
+        // aten::slice.tensor needs an additional `dim` input.
+        if (sliceable->type()->isSubtypeOf(TensorType::get())) {
+          Value* dim = method.graph()->insertConstant(0, val_range);
+          Value* sliced = emitBuiltinCall(
+              val_range,
+              *method.graph(),
+              aten::slice,
+              {sliceable, dim, psv->start(), psv->stop(), psv->step()},
+              {});
+          return std::make_shared<SimpleValue>(sliced);
+        }
+
+        Value* sliced = emitBuiltinCall(
+            val_range,
+            *method.graph(),
+            aten::slice,
+            {sliceable, psv->start(), psv->stop(), psv->step()},
+            {});
+        return std::make_shared<SimpleValue>(sliced);
```

Comment:
These branches look identical except for `dim`. Consider using a vector instead of array literals for the method args so you can reuse the `emitBuiltinCall` and `return`.

Reply:
Done.